File: examples/ios_shell_host/CrosswakeShell/CrosswakeShellConfig.swift
import Foundation

public struct CrosswakeShellConfig {
    public weak var appInfoDelegate: AppInfoDelegate?
    public weak var hapticsDelegate: HapticsDelegate?
    public weak var permissionStatusDelegate: PermissionStatusDelegate?
    public weak var notificationTokenDelegate: NotificationTokenDelegate?
    public weak var shareDelegate: ShareDelegate?
    public weak var filesPickDelegate: FilesPickDelegate?

    public init(
        appInfoDelegate: AppInfoDelegate? = nil,
        hapticsDelegate: HapticsDelegate? = nil,
        permissionStatusDelegate: PermissionStatusDelegate? = nil,
        notificationTokenDelegate: NotificationTokenDelegate? = nil,
        shareDelegate: ShareDelegate? = nil,
        filesPickDelegate: FilesPickDelegate? = nil
    ) {
        self.appInfoDelegate = appInfoDelegate
        self.hapticsDelegate = hapticsDelegate
        self.permissionStatusDelegate = permissionStatusDelegate
        self.notificationTokenDelegate = notificationTokenDelegate
        self.shareDelegate = shareDelegate
        self.filesPickDelegate = filesPickDelegate
    }
}

File: examples/android_shell_host/app/src/main/java/dev/crosswake/shell/CrosswakeShellConfig.kt
package dev.crosswake.shell

data class CrosswakeShellConfig(
    val appInfoDelegate: AppInfoDelegate? = null,
    val hapticsDelegate: HapticsDelegate? = null,
    val permissionStatusDelegate: PermissionStatusDelegate? = null,
    val notificationTokenDelegate: NotificationTokenDelegate? = null,
    val shareDelegate: ShareDelegate? = null,
    val filesPickDelegate: FilesPickDelegate? = null
)
